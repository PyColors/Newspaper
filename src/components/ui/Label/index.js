import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../../styles/theme';

const { fontSize, colors } = theme;

const Label = styled.label`
  color: ${props => props.color || `${colors.primaryTwo}`};
  outline: none;
  font-size: ${fontSize.large};
  padding: 8px 10px;
`;

const TitleLabel = styled.span`
  font-size: ${fontSize.large};
  color: ${props => props.color || `${colors.primaryTwo}`};
`;

class LabelComponent extends Component {
  render() {
    const { title, htmlFor } = this.props;

    return (
      <Label htmlFor={htmlFor}>
        <TitleLabel>
          {title}
        </TitleLabel>
      </Label>
    );
  }
}

Label.propTypes = {
  title: PropTypes.string.isRequired,
  htmlFor: PropTypes.string
};

Label.defaultProps = {
  htmlFor: ''
};
export default LabelComponent;
