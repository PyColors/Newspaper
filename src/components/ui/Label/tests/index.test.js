import React from 'react';
import { shallow } from 'enzyme';

import Label from '../index';

const title = 'title description';
const htmlFor = 'htmlFor';

const renderComponent = (props = {}) =>
  shallow(<Label title={'title'} htmlFor={'htmlFor'} {...props} />);

describe('<Label />', () => {
  it('should have an title attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('title')).toEqual(title);
  });

  it('should have an htmlFor attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('htmlFor')).toEqual(htmlFor);
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(<Label>{children}</Label>);
    expect(renderedComponent.contains(children)).toBe(false);
  });

  it('should not have a className attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('className')).toBeUndefined();
  });

  it('should assume a className attribute', () => {
    const className = 'test';
    const renderedComponent = renderComponent({ className });
    expect(renderedComponent.hasClass(className)).toBe(true);
  });
});
