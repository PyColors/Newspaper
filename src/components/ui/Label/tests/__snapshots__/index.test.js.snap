// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should match the snapshot 1`] = `
<label
  className="sc-bdVaJa kYKWkt"
  htmlFor="PyColors"
  title="Title Label"
>
  <span
    className="sc-bwzfXH lnTsJm"
  >
    What do you think
  </span>
</label>
`;
